datajunction-clients/python/datajunction/builder.py

```diff
@@ -238,8 +283,21 @@ def register_table(self, catalog: str, schema: str, table: str) -> Source:
                 f"Failed to register table `{catalog}.{schema}.{table}`: {exc}",
             ) from exc
         source_node = Source(
-            **response.json(),
             dj_client=self,
+            name=response.json()["name"],
```

Comment:
Might be better to serialize to json once, like `data = response.json()` then do `data["name"]` to pull out the values.